uniparc-common/src/main/java/org/uniprot/api/uniparc/common/service/light/UniParcCrossReferenceService.java

```diff
@@ -154,6 +172,12 @@ private Stream<UniParcCrossReference> getFilteredCrossReferences(
                 .filter(xref -> filterCrossReference(xref, request));
     }
 
+    private Stream<UniParcCrossReference> getFilteredDatabaseCrossReferences(
+            UniParcEntryLight uniParcEntryLight, UniParcDatabasesRequest request) {
+        return getFilteredCrossReferences(uniParcEntryLight,request)
+                .filter(xref -> new UniParcDatabaseIdFilter().apply(xref, request.getId()));
+    }
```

Comment:
No need to create filter every time. 
Create only one object of type new UniParcDatabaseIdFilter() and reuse it

Reply:
I think, we should ignore case while comparing. q12345 should match Q12345